### PR TITLE
#5560 rewrite the plot domain mapping, range and axis

### PR DESCRIPTION
**Pull request by MariuszJurowicz**

### CODE REVIEW

**Review by piorek — APPROVED** [review]

File: js/notebook/src/plot/plotScope.js [hunk 1]
```diff
@@ -262,16 +262,20 @@ define([
       this.plotFocus.focus
     );
 
-    this.calcMapping(false);
+    this.plotRange.calcMapping(false);
     this.update();
   };
 
+  PlotScope.prototype.calcMapping = function(emitFocusUpdate) {
```
Comment (piorek) [inline js/notebook/src/plot/plotScope.js:269]: @mjurowicz-xsolve Is it used anywhere? It looks like it's not needed anymore. But maybe I'm wrong.

File: js/notebook/src/plot/std/axis/TimeAxis.ts [hunk 2]
```diff
@@ -0,0 +1,349 @@
...
+  setRange(vl, vr, para): void {
+    if (vl !== null) { this.axisValL = vl; }
+    if (vr !== null) { this.axisValR = vr; }
+
+    this.setTimeAxisRange(para);
+
+    this.axisValSpan = plotUtils.minus(this.axisValR, this.axisValL);
+  }
+
+  setTimeAxisRange(para) {
```
Comment (piorek) [inline js/notebook/src/plot/std/axis/TimeAxis.ts]: I suggest a meaningful name like axisTimezone perhaps?

File: js/notebook/src/plot/std/axis/DefaultAxis.ts [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+    if (vr !== null) { this.axisValR = vr; }
+
+    if (this.axisType === "log") {
+      this.setLogAxisBase(para);
+    }
+
+    this.axisValSpan = plotUtils.minus(this.axisValR, this.axisValL);
+  }
+
+  setLogAxisBase(para) {
```
Comment (piorek) [inline js/notebook/src/plot/std/axis/DefaultAxis.ts]: para -> axisBase

File: js/notebook/src/plot/plotScope.js (hunk 1 above)
Comment (MariuszJurowicz) [reply js/notebook/src/plot/plotScope.js:269]: It's used in `CombinedPlots` yet

File: js/notebook/src/plot/std/axis/TimeAxis.ts (hunk 2 above)
Comment (MariuszJurowicz) [reply js/notebook/src/plot/std/axis/TimeAxis.ts]: Yes, I've missed that

File: js/notebook/src/plot/std/axis/DefaultAxis.ts (hunk 3 above)
Comment (MariuszJurowicz) [reply js/notebook/src/plot/std/axis/DefaultAxis.ts]: Yes, I've missed that

**Pull request merged**